src/main/kotlin/org/web3j/quorum/enclave/Constellation.kt

```diff
@@ -28,14 +28,18 @@ class Constellation(private val ipcService: EnclaveService, private val web3: Qu
         return web3.ethSendRawTransaction(payload).send()
     }
 
+    // payload needs to be base64 encoded string
+    // returned hash is base64 encoded
     override fun storeRawRequest(payload: String, from: String, to: List<String>): SendResponse {
         val sendRequest = SendRequest(payload, from, to)
         return ipcService.send(sendRequest, "send", SendResponse::class.java)
     }
 
-    override fun receiveRequest(key: String, to: String): ReceiveResponse {
+    // key is base64 encoded directly returned from constellation storeRawRequest
+    // returned string is base64 encoded payload
+    override fun receiveRequest(key: String, to: String): String {
         val receiveRequest = ReceiveRequest(key, to)
-        return ipcService.send(receiveRequest, "receive", ReceiveResponse::class.java)
+        return ipcService.send(receiveRequest, "receive", ReceiveResponse::class.java).payload
     }
```

Comment:
Has this really changed in Constellation?

Reply:
Nothing in Constellation has changed, but the ReceiveRequest return type in Enclave.kt was changed from ReceiveResponse to String to allow it to work with Tessera call